packages/server/lib/controllers/sync/deploy/postConfirmation.ts

```diff
@@ -42,7 +44,8 @@ export const postDeployConfirmation = asyncWrapper<PostDeployConfirmation>(async
     if (body.onEventScriptsByProvider) {
         const diff = await onEventScriptService.diffChanges({
             environmentId,
-            onEventScriptsByProvider: body.onEventScriptsByProvider
+            onEventScriptsByProvider: body.onEventScriptsByProvider,
```

Comment:
I am actually confused why `onEventScriptsByProvider` would contain events script for integration that are not being deployed. Does that mean we have a bug in the deploy.service package function that should not have included scripts that don't belong to the specific integration. Can we ensure the CLI is only actually actually sending the necessary data, instead of trying to fix it server side?

Reply:
got it but by having the CLI only send the appropriate on events scripts, we could avoid having to filter them here https://github.com/NangoHQ/nango/pull/4037/files#diff-4fae00b45e13057c41ce811e773605e86d4816ff08bc9d499a3237531863319bR180-R182